docs/user/secure-gateways.md

```diff
@@ -0,0 +1,237 @@
+# Secure Gateways
+
+This guide will help you get started using secure Gateways. The guide uses a self-signed CA, so it should be used for
+testing and demonstration purposes only.
+
+## Prerequisites
+
+- A Kubernetes cluster with `kubectl` context configured for the cluster.
+- OpenSSL to generate TLS assets.
+
+__Note:__ Envoy Gateway is tested against Kubernetes v1.24.
+
+## Installation
+
+Follow the steps from the [Quickstart Guide](QUICKSTART.md) to install Envoy Gateway and the example manifest.
+Before proceeding, you should be able to query the example backend using HTTP.
+
+## TLS Certificates
+
+Generate the certificates and keys used by the Gateway to terminate client TLS connections.
+
+Create a root certificate and private key to sign certificates:
+
+```shell
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+```
+
+Create a certificate and a private key for `www.example.com`:
+
+```shell
+openssl req -out www.example.com.csr -newkey rsa:2048 -nodes -keyout www.example.com.key -subj "/CN=www.example.com/O=httpbin organization"
+openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in www.example.com.csr -out www.example.com.crt
+```
+
+Store the cert/key in a Secret:
+
+```shell
+kubectl create secret tls example-cert --key=www.example.com.key --cert=www.example.com.crt
+```
+
+Update the Gateway from the Quickstart guide to include an HTTPS listener that listens on port `8443` and references the
+`example-cert` Secret:
+
+```console
+$ cat <<EOF | kubectl apply -f -
```

Comment:
a `patch` to the listener on an exisiting quickstart Gateway might be better.

Reply:
did this for the tls-passthrough docs PR https://github.com/envoyproxy/gateway/pull/575/files#diff-c9d477931ae33d7a5bcbf71cf85dd9ca1d8fbb09e13367765df70fa00c5dc182R58

```console
$ kubectl patch gateway eg --type=json --patch '[{
   "op": "add",
   "path": "/spec/listeners/-",
   "value": {
      "name": "tls",
      "protocol": "TLS",
      "hostname": "passthrough.example.com",
      "tls": {"mode": "Passthrough"}, 
      "port": 6443,
    },
}]'
```